### PR TITLE
Use Stdio instead of HTTP

### PR DESCRIPTION
**Pull request by david-driscoll**
Work in progess with a couple remaining todos... see #113

### CODE REVIEW

File: lib/omni-sharp-server/omni-sharp-server.ts [hunk 1]
```diff
@@ -60,54 +61,138 @@ atom.emitter.on("omni-sharp-server:state-change", (state) => {
...
+    public child: child_process.ChildProcess;
     public port;
+    private _seq = 0;
 
     public start() {
         var executablePath, useMono;
         useMono = process.platform !== "win32";
         executablePath = this.location;
-        return findFreePort((err, port) => {
+        findFreePort((err, port) => {
```
Comment (nosami) [inline lib/omni-sharp-server/omni-sharp-server.ts]: guess we don't need to find a free port any more :)

Comment (ctolkien) [inline lib/omni-sharp-server/omni-sharp-server.ts]: Can remove the whole dependency on findFreePort

File: lib/omni-sharp-server/omni-sharp-server.ts [hunk 2]
```diff
@@ -60,54 +61,138 @@ atom.emitter.on("omni-sharp-server:state-change", (state) => {
...
+    private _partialResult: string;
+
+    // TODO: Review if this is the best way...
+    private serverResponse = (data: any) => {
+        var result :string = data.toString();
+        if (this._partialResult) {
+            result = this._partialResult + result;
+        }
+
+        var results = result.split('\n');
```
Comment (nosami) [inline lib/omni-sharp-server/omni-sharp-server.ts]: readline https://nodejs.org/api/readline.html would probably simplify this code.

Comment (david-driscoll) [inline lib/omni-sharp-server/omni-sharp-server.ts]: Well then...

File: lib/omni-sharp-server/omni-sharp-server.ts (hunk 1 above)
Comment (david-driscoll) [inline lib/omni-sharp-server/omni-sharp-server.ts]: I wasn't sure, so I left it for the moment.

**Comment by david-driscoll** [comment]
Updated @nosami @ctolkien @willl

**Comment by david-driscoll** [comment]
Rebased onto master :+1:

**Comment by nosami** [comment]
screen shot 2015-04-24 at 05 44 57

I get this false diagnostic on each file.

**Pull request merged**